test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,224 @@
+                    if (isset($this->specs[$key])) {
+                        return $this->specs[$key];
+                    }
+
+                    $item              = $callback(new class () implements ItemInterface {
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function getKey()
+                        {
+                            return '';
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function set($value)
+                        {
+                            return $this;
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function expiresAfter($time)
+                        {
+                            return $this;
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function isHit()
+                        {
+                            return true;
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function get()
+                        {
+                            return null;
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function getMetadata(): array
+                        {
+                            return [];
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function tag($tags): ItemInterface
+                        {
+                            Assert::assertSame(SpecificationLoader::CACHE_TAG, $tags);
+
+                            return $this;
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function expiresAt($expiration)
+                        {
+                            return $this;
+                        }
+                    });
+                    $this->specs[$key] = $item;
+
+                    return $item;
+                }
+
+                /**
+                 * {@inheritDoc}
+                 */
+                public function invalidateTags(array $tags)
+                {
+                    return true;
+                }
+
+                public function delete(string $key): bool
+                {
+                    return true;
+                }
+            }
+        );
+
+        parent::setUp();
+    }
+
+    public function testRegisterSpecAndGettingRegisteredSpecification(): void
```

Comment:
The exception thrown when calling `get` with a specification name that was not set should be also tested in a separate test method.